src/Interpreters/ClientInfo.h

```diff
@@ -101,6 +102,14 @@ class ClientInfo
     /// The element can be trusted only if you trust the corresponding proxy.
     /// NOTE This field can also be reused in future for TCP interface with PROXY v1/v2 protocols.
     String forwarded_for;
+    String getLastForwardedFor() const
+    {
+        if (forwarded_for.empty())
+            return {};
+        String last = forwarded_for.substr(forwarded_for.find_last_of(',') + 1);
```

Comment:
Why don't do it once when we set `forwarded_for`?

Reply:
we set it in different places - depends on protocol, and we don't use it often - so no need to keep it as a dedicated field